src/map/battle.cpp

```diff
@@ -9031,6 +9164,45 @@ enum damage_lv battle_weapon_attack(struct block_list* src, struct block_list* t
 	if (sd && tsc && wd.flag&BF_LONG && tsc->data[SC_WINDSIGN] && rand()%100 < tsc->data[SC_WINDSIGN]->val2)
 		status_heal(src, 0, 0, 1, 0);
 
+	if (sc && sc->data[SC_AUTO_FIRING_LAUNCHER])
+	{
+		switch (sc->data[SC_AUTO_FIRING_LAUNCHER]->val1)
```

Comment:
Applies to all below:
Should use the skill_lv lookup value and dont look it up twice.